lib/trento/domain/cluster/cluster.ex

```diff
@@ -204,6 +205,48 @@ defmodule Trento.Domain.Cluster do
   def execute(%Cluster{cluster_id: nil}, _),
     do: {:error, :cluster_not_registered}
 
+  # Restoration, when a RegisterClusterHost command is received for a deregistered Cluster
+  # the cluster is restored, the host is added to cluster and if the host is a DC
+  # cluster details will be updated
+  def execute(
+        %Cluster{deregistered_at: deregistered_at, cluster_id: cluster_id},
+        %RegisterClusterHost{
+          host_id: host_id,
+          designated_controller: false
+        }
+      )
+      when not is_nil(deregistered_at) do
+    [
+      %ClusterRestored{cluster_id: cluster_id},
+      %HostAddedToCluster{
+        cluster_id: cluster_id,
+        host_id: host_id
+      }
+    ]
+  end
+
+  def execute(
+        %Cluster{deregistered_at: deregistered_at, cluster_id: cluster_id} = cluster,
+        %RegisterClusterHost{
+          host_id: host_id,
+          designated_controller: true
+        } = command
+      )
+      when not is_nil(deregistered_at) do
+    cluster
+    |> Multi.new()
+    |> Multi.execute(fn _ ->
+      %ClusterRestored{cluster_id: cluster_id}
+    end)
+    |> Multi.execute(fn _ ->
```

Comment:
The `HostAddedToCluster` is already considered in the `maybe_update_cluster`.
I think that if you remove this, the tests will even pass the same way

Reply:
Yes absolutely, I left on purpose to be consistent with the non dc restoration, I don't want to hide this is the maybe update, for the purpose of the flow is a no-op no double events emitted, for the reader it's immediate to understand the flow by looking at both clauses, wdyt?

Reply:
I want to explicit say that the host is always added to cluster and not hide the details in the maybe for the reader (the final output is the same) just for the sake of readability

Reply:
Alright, no strong opinions.